include/aspect/mesh_deformation/interface.h

```diff
@@ -228,7 +228,7 @@ namespace aspect
          * mesh deformation object. If no free surface is used,
          * an empty set is returned.
          */
-        const std::set<types::boundary_id>
+        const std::set<types::boundary_id> &
```

Comment:
Ouch. This was the reason the `find` function failed? The call created two different copies of the same object so the returned iterator of the one object was never the `end` iterator of the other one? Nice find. Could you try removing the `tmp` objects you created and see if it still works?

Reply:
Yep, that was it. It still works without the `tmp` objects, but I thought it was neater to just ask for the `free_surface_boundary_indicators` once outside the loop. Do you agree?

Reply:
It looks a bit nicer, yes. But then please make the `tmp` objects into references as well, otherwise you copy the whole set anyway. Like this:
```
const std::set<types::boundary_id> &tmp_free_surface_boundary_indicators = this->...
```